community/values/src/main/java/org/neo4j/values/storable/CoordinateReferenceSystem.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values.storable;
+
+import org.neo4j.graphdb.spatial.CRS;
+
+public class CoordinateReferenceSystem implements CRS
+{
+    public static final CoordinateReferenceSystem Cartesian = new CoordinateReferenceSystem( "cartesian", CRSTable.SR_ORG, 7203 );
+    public static final CoordinateReferenceSystem WGS84 = new CoordinateReferenceSystem( "WGS-84", CRSTable.EPSG, 4326 );
+
+    public static CoordinateReferenceSystem get( int tableId, int code )
+    {
+        CRSTable table = CRSTable.find( tableId );
+        // TODO this might break, since we are changing the name. Use the function below?
+        return new CoordinateReferenceSystem( table.name + "-" + code, table, code );
```

Comment:
not sure why do we need to create new instances of reference system here? we have only fixed number of those systems, right?